packages/web-config-server/src/authSession/authSession.js

```diff
@@ -1,45 +1,46 @@
 import {} from 'dotenv/config'; // Load the environment variables into process.env
 import session from 'client-sessions';
 
+import { getUserFromBasicAuth } from './getUserFromBasicAuth';
 import { getAccessPolicyForUser } from './getAccessPolicyForUser';
 import { PUBLIC_USER_NAME } from './publicAccess';
 
 const allowedUnauthRoutes = ['/login', '/version'];
 
 // auth is a middleware that runs on every request
-const auth = () => (req, res, next) => {
-  if (
-    (req.session && req.session.userJson && req.session.userJson.userName) ||
-    checkAllowedUnauthRoutes(req)
-  ) {
-    // if logged in or loggin in continue
+const auth = () => async (req, res, next) => {
+  // if using basic auth, check credentials and set access policy for that user
+  const basicAuthUser = await getUserFromBasicAuth(req);
+  if (basicAuthUser) {
+    req.accessPolicy = await getAccessPolicyForUser(req.models, basicAuthUser.fullName);
     next();
-    // else check if this is the first request after user logged out
-    // and send 440 or authenticate as public user
-  } else if (req.lastuser && req.lastuser.userName && req.lastuser.userName !== PUBLIC_USER_NAME) {
+    return;
+  }
+
+  // if logged in or logging in continue
+  const userName = req.session?.userJson?.userName;
+  if (!!userName || checkAllowedUnauthRoutes(req)) {
+    req.accessPolicy = req.accessPolicy || (await getAccessPolicyForUser(req.models, userName));
+    next();
+    return;
+  }
+
+  // check if this is the first request after user logged out and send 440
+  if (req.lastuser?.userName !== PUBLIC_USER_NAME) {
```

Comment:
I think the control flow is slightly changed here? Previously if `req.lastUser.userName = undefined` we would fail this if check whereas now we succeed? Not sure if that's at all a valid case though...

Reply:
I think it should still fail, because `PUBLIC_USER_NAME` is not `undefined` :-)

Reply:
Oooops... 🤦

Reply:
Yep you're totally right, I mucked that up!

Reply:
Fixed!